File: Mapper/Exception/NotSameTypePropertyException.vb
Namespace Exception
    Public Class NotSameTypePropertyException
        Inherits System.Exception

        Public Sub New(typeSource As Type, typeDest As Type)
            MyBase.New("Les propriétés source et de destination ne sont pas du même type(source est de type " + typeSource.Name + " et destination est de type " + typeDest.Name + ")")
        End Sub
    End Class
End Namespace

File: MapperCore/Exception/NotSameTypePropertyException.cs

using System;
namespace MapperCore.Exception
{
    public class NotSameTypePropertyException : System.Exception
    {

        public NotSameTypePropertyException(Type typeSource, Type typeDest)
            : base("Les propriétés source et de destination ne sont pas du même type(source est de type " + typeSource.Name + " et destination est de type " + typeDest.Name + ")")
        {

        }
    }
}
